python/paddle/fluid/optimizer.py

```diff
@@ -388,13 +388,14 @@ def minimize(self,
             for param in parameters:
                 if not param.trainable:
                     continue
-                # create gradient variable
-                grad_var = Variable(
-                    block=loss.block,
-                    name=param._ivar._grad_name(),
-                    stop_gradient=True,
-                    ivar=param._ivar._grad_ivar())
-                params_grads.append((param, grad_var))
+                if param._ivar._grad_ivar() is not None:
```

Comment:
why?

Reply:
Here we will get all params to minimize, however some param shouldn't be minimized cuz it has been overwritten by other params. So we should avoid creating wrong grads for those params